src/pollution_model/run_pollution_model.py

```diff
@@ -0,0 +1,454 @@
+                                                                   average_rain_intensity=average_rain_intensity,
+                                                                   event_duration=event_duration,
+                                                                   rainfall_ph=rainfall_ph, surface_type=surface_type)
+        curr_dissolved_copper, curr_dissolved_zinc = dissolved_metal_load(total_copper_load=curr_total_copper,
+                                                                          total_zinc_load=curr_total_zinc,
+                                                                          surface_type=surface_type)
+        all_buildings.loc[{'row': i}] = [all_buildings.loc[{'row': i}][0], surface_area, surface_type, curr_tss, curr_total_copper, curr_total_zinc, curr_dissolved_copper, curr_dissolved_zinc]
+        # print(all_buildings.loc[{'row': i, 'col': "TSS"}])
+        # all_buildings.loc[{'row': i, 'col': 'TSS'}] = curr_tss
+        # all_buildings.loc[{'row': i, 'col': 'TCu'}] = curr_total_copper
+        # all_buildings.loc[{'row': i, 'col': 'TZn'}] = curr_total_zinc
+        # all_buildings.loc[{'row': i, 'col': 'DCu'}] = curr_dissolved_copper
+        # all_buildings.loc[{'row': i, 'col': 'DZn'}] = curr_dissolved_zinc
+    print(all_buildings)
+    # Run through all the roads/car parks, and calculate TSS, total metal loads, and dissolved metal loads
+    for i in range(len(all_roads)):
+        surface_area = float(all_buildings.sel(row=i, col="SurfaceArea"))
+        surface_type = int(all_buildings.sel(row=i, col="SurfaceType"))
+        curr_tss = compute_tss_roof_road(surface_area=surface_area, antecedent_dry_days=antecedent_dry_days,
+                                         average_rain_intensity=average_rain_intensity, event_duration=event_duration,
+                                         surface_type=surface_type)
+
+        curr_total_copper, curr_total_zinc = total_metal_load_road_carpark(curr_tss)
+        curr_dissolved_copper, curr_dissolved_zinc = dissolved_metal_load(total_copper_load=curr_total_copper,
+                                                                          total_zinc_load=curr_total_zinc,
+                                                                          surface_type=surface_type)
+
+        all_roads.loc[{'row': i, 'col': 'TSS'}] = curr_tss
+        all_roads.loc[{'row': i, 'col': 'TCu'}] = curr_total_copper
+        all_roads.loc[{'row': i, 'col': 'TZn'}] = curr_total_zinc
+        all_roads.loc[{'row': i, 'col': 'DCu'}] = curr_dissolved_copper
+        all_roads.loc[{'row': i, 'col': 'DZn'}] = curr_dissolved_zinc
+    print(all_roads)
+
+    all_result = xr.merge(all_roads, all_buildings)
+
+def main(selected_polygon_gdf: gpd.GeoDataFrame,
+         log_level: LogLevel = LogLevel.DEBUG,
+         antecedent_dry_days: float = 1,
+         average_rain_intensity: float = 1,
+         event_duration: float = 1,

```

Comment:
Might be useful to have the units in the variable names here, what are your thoughts?

Reply:
I'm not so sure. I think that saying "days" or "duration" indicates that this is a measurement of time, and "rain"intensity" is also descriptive. The units are included in the docstring of the function as well.